### PR TITLE
Minor core gas reductions

### PR DESCRIPTION
**Pull request by ryley-o**
Implement minor core gas optimizations to slightly reduce minting costs.

This finds ~0.4% gas reduction on mints via:
- identifying areas where safe math is not required 
- eliminating an unnecessary `address(0)` assignment when no funds are due to artist or additional payee on primary sales

Changes are all internal and do no affect interfaces.

## Gas impacts
Mint costs are slightly reduced. Representative minter cost updates are shown in the table below.

_All gas costs assume project 3 mints avg(501-516) of 1000_
| Gas Test | Previous gas used to purchase | New gas used to purchase | % change |
| --- | --- | --- | --- |
| MinterSetPriceV2_V3Core | 138407 | 137859 | -0.4% (cheaper) |
| MinterDAExpV2_V3Core | 148156 | 147608 | -0.4% (cheaper) |

## Audit impacts
- Highly recommend this is included in audit

### CODE REVIEW

File: contracts/GenArt721CoreV3.sol [hunk 1]
```diff
@@ -259,7 +262,10 @@ contract GenArt721CoreV3 is ERC721, Ownable, IGenArt721CoreContractV3 {
         // EFFECTS
         // increment project's invocations
         projects[_projectId].invocations = invocationsAfter;
-        uint256 thisTokenId = (_projectId * ONE_MILLION) + invocationsBefore;
+        uint256 thisTokenId;
+        unchecked {
+            thisTokenId = (_projectId * ONE_MILLION) + invocationsBefore;
```
Comment (jakerockland) [inline contracts/GenArt721CoreV3.sol:267]: Same feedback here re: comment.

File: contracts/GenArt721CoreV3.sol [hunk 2]
```diff
@@ -1170,20 +1176,30 @@ contract GenArt721CoreV3 is ERC721, Ownable, IGenArt721CoreContractV3 {
     {
         // calculate revenues
         artblocksRevenue_ = (_price * artblocksPrimarySalesPercentage) / 100;
-        uint256 projectFunds = _price - artblocksRevenue_;
+        uint256 projectFunds;
+        unchecked {
+            // artblocksRevenue_ is always <=25, so guaranteed to never underflow
+            projectFunds = _price - artblocksRevenue_;
```
Comment (jakerockland) [inline contracts/GenArt721CoreV3.sol:1182]: Same feedback here re: comment.

Comment (jakerockland) [inline contracts/GenArt721CoreV3.sol:1182]: Ty for comment here :)

File: contracts/GenArt721CoreV3.sol [hunk 3]
```diff
@@ -1170,20 +1176,30 @@ contract GenArt721CoreV3 is ERC721, Ownable, IGenArt721CoreContractV3 {
...
+        }
         additionalPayeePrimaryRevenue_ =
             (projectFunds *
                 projectIdToAdditionalPayeePrimarySalesPercentage[_projectId]) /
             100;
-        artistRevenue_ = projectFunds - additionalPayeePrimaryRevenue_;
+        unchecked {
+            // projectIdToAdditionalPayeePrimarySalesPercentage is always
+            // <=100, so guaranteed to never underflow
+            artistRevenue_ = projectFunds - additionalPayeePrimaryRevenue_;
```
Comment (jakerockland) [inline contracts/GenArt721CoreV3.sol:1191]: Ty for comment here :)

File: contracts/GenArt721CoreV3.sol [hunk 4]
```diff
@@ -238,7 +238,10 @@ contract GenArt721CoreV3 is ERC721, Ownable, IGenArt721CoreContractV3 {
 
         // load invocations into memory
         uint24 invocationsBefore = projects[_projectId].invocations;
-        uint24 invocationsAfter = invocationsBefore + 1;
+        uint24 invocationsAfter;
+        unchecked {
+            invocationsAfter = invocationsBefore + 1;
```
[the remark this answers is not in the document]
Comment (ryley-o) [reply contracts/GenArt721CoreV3.sol:245]: agreed and updated!

File: contracts/GenArt721CoreV3.sol (hunk 1 above)
Comment (ryley-o) [reply contracts/GenArt721CoreV3.sol:273]: updated!

**Pull request merged**